### ISSUE TITLE
[Question] Content-Type issue with GraphQL?

### ISSUE THREAD

**Issue by JustinTBrown** [getinsomnia/insomnia#755]
- Insomnia Version: 5.12.4.1752
- Operating System: MacOS 10.12.6

## Details

With GraphQL queries, I assume I need to specify the Content-Type header as `application/graphql` since I'm providing the query in the GraphQL language:

```
query {
  readQuickPossessions {
    edges {
      node {
        ID
        Title
        Address
        SquareFeet
      }
    }
  }
}
```
However this query results in an error:

```
{
	"data": null,
	"errors": [
		{
			"message": "Syntax Error GraphQL request (1:1) Unexpected <EOF>\n\n1: \n   ^\n",
			"locations": [
				{
					"line": 1,
					"column": 1
				}
			]
		}
	]
}
```

Instead, if I change the content-type header to `application/json` the query works:

```
{
	"data": {
		"readQuickPossessions": {
			"edges": [
				{
					"node": {
						"ID": "1",
						"Title": "Bellagio",
						"Address": "123 Smith St.",
						"SquareFeet": 1787
					}
				},
				{
					"node": {
						"ID": "2",
						"Title": "Tremblay",
						"Address": "456 Jenkins Rd.",
						"SquareFeet": 2192
					}
				}
			]
		}
	}
}
```

Any idea what might be happening here?

**Comment by gschier** [comment]
Hmm, Insomnia uses `application/json` to send GraphQL requests, which is what it should be.

Only the query part of a GraphQL request is `application/graphql` but it's wrapped in a JSON object, which is why `application/json` is used. Here's an example to demonstrate what a raw GraphQL request looks like:

```json
{
  "query" : "{"query":"query myQuery {\n\tfoo\n\tbar\n}"}",
  "operationName": "myQuery",
  "variables": {
    "my": "variable"
  }
}
```

**Comment by JustinTBrown** [comment]
Ok, that makes sense! Thanks for the clarification. I wasn't distinguishing correctly between the request and the query. Thanks.

**Comment by gschier** [comment]
No problem 😄

**Comment by RobertWSaunders** [comment]
> @gschier Only the query part of a GraphQL request is application/graphql but it's wrapped in a JSON object, which is why application/json is used.

I'm not sure if the docs have been updated since your comment but I believe when using the `application/graphql` content type the GraphQL operation should be sent as plain text in the body of the request and not wrapped in a JSON object. As a result, Insomnia currently does not support the `application/graphql` content type.

I verified this against an API that supports both `application/json` and `application/graphql` content types, when using `application/graphql` with Insomnia the server responds with (similar to @JustinTBrown response):

```
{
  "errors": [
    {
      "message": "Parse error on \"query\" (STRING) at [1, 2]",
      "locations": [
        {
          "line": 1,
          "column": 2
        }
      ]
    }
  ]
}
```

However when using `application/graphql` with cURL and passing the GraphQL operation in the request body the requests succeeds as expected.

Can we re-open this issue?
